app/code/Magento/Customer/Model/Customer.php

```diff
@@ -883,6 +883,12 @@ public function sendPasswordResetConfirmationEmail()
      */
     public function getGroupId()
     {
+		if ($this->hasData('id') && !$this->hasData('group_id')) {
+			 $groupId=$customerDataObject= $this->_getResource()->getGroupId($this->getData('id'));			 
+			 if($groupId) {
+				 $this->setData('group_id', $groupId);
```

Comment:
Getter must not change object state.

Reply:
I have remove $customerDataObject= is it ok